File: blood-bank-be/bloodBank/src/main/java/ftn/uns/ac/rs/bloodbank/registration/dto/AddressRequest.java
package ftn.uns.ac.rs.bloodbank.registration.dto;

import lombok.*;

@Getter
@AllArgsConstructor
@EqualsAndHashCode
@ToString
@NoArgsConstructor
public class AddressRequest {
    private String city;
    private String street;
    private String country;
    private String streetNumber;

}

File: blood-bank-fe/src/app/model/Requests/AddressRequest.ts
export interface IAddressRequest {
    city?: string,
    street?: string,
    country?: string,
    streetNumber?: string
  }

export class AddressRequest implements IAddressRequest {
  city?: string;
  street?: string;
  country?: string;
  streetNumber?: string;
  constructor(data?: IAddressRequest) {
    if (data) {
      for (const property in data) {
        if (data.hasOwnProperty(property))
          (<any>this)[property] = (<any>data)[property];
      }
    }

  }
}
